### ISSUE TITLE
OrphanRemoval does not work when clearing uninitialized collection

### ISSUE THREAD

**Issue by akomm** [doctrine/mongodb-odm#1497]
Having Document A and B: A has ReferenceMany to B and orphanRemoval=true. 

1. When I fetch A 
2. Clear its ref-collection to B
3. Persist A

Then B items remain in the database, because the collection was not initialized so its referenced documents (B) are NOT scheduled for orphan removal:

[link](https://github.com/doctrine/mongodb-odm/blob/d728cc1e309475ed52f4cd9aa35208e699941f9e/lib/Doctrine/ODM/MongoDB/PersistentCollection/PersistentCollectionTrait.php#L538-L546)

**Comment by Steveb-p** [comment]
```
if ($this->initialized && $this->isEmpty()) {
    return;
}
```

`$this->isEmpty()` invokes initialization. (see [count method](https://github.com/doctrine/mongodb-odm/blob/d728cc1e309475ed52f4cd9aa35208e699941f9e/lib/Doctrine/ODM/MongoDB/PersistentCollection/PersistentCollectionTrait.php#L437-L453)).

Did you by any chance forgot to actually `flush` the manager?

**Comment by akomm** [comment]
No. It is an existing bug. See, if it is not initialized, `$this->isEmpty()` will never be invoked. Just ensured with debugger, but it is obvious it does not.

**Comment by malarzm** [comment]
@akomm thanks for the report! Any chance you could hack together a failing test case? It'd help a lot :)

**Comment by akomm** [comment]
Done

**Comment by alcaeus** [comment]
Fixes have been merged in #1498 and #1500 - closing this.
